base/src/main/proto/spine/options.proto

```diff
@@ -180,59 +180,49 @@ extend google.protobuf.FieldOptions {
     //
     // The value of the option is the name of the target event message/context field
     // which is used to set the corresponding enrichment's field.
-    // The value must have event/context FQN prefix if `enrichment_for` option is omitted
-    // ("example.events.MyEvent.user_id" instead of just "user_id").
+    // The value must have event/context plain field name (i.e. FQN names are not supported).
     //
     // An example:
     //
     // message MyEvent {
-    //    spine.base.UserId user_id = 1;
-    //    ...
-    //    message Enrichment {
-    //       spine.people.PersonName name = 1 [(by) = "user_id"];
-    //
-    //       string time = 2 [(by) = "context.timestamp"];
-    //    }
+    //     spine.base.UserId user_id = 1;
     // }
     //
     // message MyEventEnrichment {
-    //     string username = 1 [(by) = "MyEvent.user_id"];
-    // }
+    //     option (enrichment_for) = "MyEvent";
     //
-    // If a single enrichment message is used to enrich several events, a wildcard syntax may be
-    // used.
-    //
-    // An example:
-    //
-    // message EnrichmentForSeveralEvents {
-    //     string username = 1 [(by) = "*.user_id"];
+    //     string username = 1 [(by) = "user_id"];
     // }
     //
-    // message EventOne {
-    //     option (enrichment) = "EnrichmentForSeveralEvents";
+    // Also, the multiple arguments are supported. The syntax is:
     //
-    //     int32 user_id = 1;
+    // message FirstEvent {
+    //     spine.base.UserId user_id = 1;
     // }
     //
-    // message EventTwo {
-    //     option (enrichment) = "EnrichmentForSeveralEvents";
+    // message SecondEvent {
+    //     string id = 1;
     //
-    //     int32 user_id = 1;
+    //     spine.base.UserId admin_id = 1;
     // }
     //
-    // Also, the multiple arguments are supported. The syntax is:
-    //
     // message EnrichmentForSeveralEvents {
-    //     option (enrichment) = "EnrichmentForSeveralEvents";
+    //     option (enrichment_for) = "FirstEvent,SecondEvent";
     //
-    //     int32 user_id = 1 [(by) = "qualifier.One.target_field | qualifier.Two.alternate_target_field"];
+    //     string username = 1 [(by) = "user_id | admin_id"];
     // }
     //
-    // Wildcard syntax for the target fields is supported in cases when annotation `enrichment_for`
-    // gives a not ambiguous definition for the target type.
+    // If there is a complex (by) case, it is possible to supply a FQN of a rule that determines
+    // by which field the message should be enriched. The feature is experimental and can be
+    // changed without any notices.
+    //
+    // An example:
+    //
+    // message MyEnrichment {
+    //     option (enrichment_for) = "example.events.*"
     //
-    // If using the FQN field names, the `enrichment_for` annotation may be omitted
-    // (the same behavior as with a single-argument `by` annotation).
+    //     string username = 1 [(by) = "rule:com.acme.UsernameEnrichmentRule"
```

Comment:
Let's not give this in documentation until we finally agreed on how it's going to work (if at all).